### DISCUSSION TITLE
ROSCO tuning hyperparameters

### DISCUSSION THREAD

**Discussion by C-Giorgio** [NatLabRockies/ROSCO#132] [General]
Hi everyone,
I am trying to verify from the time response of the turbine the controller parameters that are set in the .yaml file for the pitch controller. I am testing the monopile IEA-15-240-RWT turbine and the desired controller parameters are 1.0 for damping and 0.2 rad/s for the natural frequency (as reported in the .yaml file). To observe this parameters in the rotor speed response, a step input on the reference signal should be applied. Since adding a step input on the reference on the DISCON.IN file is not possible (as far as I know), I run OpenFAST with the Simulink controller. I keep the wind constant at 15 m/s and after 100 s of transient, I apply a step input on the reference generator rotational speed of 0.2 rad/s (red curve). The blue curve is the response of the rotational speed. However, as the figure shows, the system is responding as an underdumped system: large overshoot and few oscillations before reaching the steady-state value. However, since the desired dumping with which the gains are computed is critical, I would expect a response without oscillations and overshoot (more similar to the green line). I know that the desired controller parameters are used to compute the gains schedule on a linearized model and what I am testing is a full non-linear model, however I believe the difference is way too large. Is there anything else that I am missing for which this type of analysis is wrong?

![Step_input_2](https://user-images.githubusercontent.com/105369849/167879919-5785c8bb-3630-4ef5-902d-56bc7cf122ee.jpg)

Thanks in advance. 

Best, Giorgio

**Comment by dzalkind** [comment]
I think my previous answer still stands: 

I'm not entirely sure what you are trying to analyze, but this response looks reasonable. If you'd like a different response, you can tune the damping (zeta_pc) and natural frequency (omega_pc). If you're looking for a more traditional second-order response, I would try a smaller step and avoid the nonlinear near-rated region when blade pitch is near 0.

Another thing I see when looking at this is the linear changes in blade pitch, which makes me think the pitch signal is rate-limited.  If you remove blade pitch rate limits the system will be more linear (as will a smaller step).  Technically, the pitch controller is gain scheduled to have a consistent response when the blade pitch changes, not the rotor speed operating point, so that could be another source of nonlinearity.  

In any case, we don't typically expect a perfect second-order response because the system isn't really second-order.  We use these parameters to tune for a desirable generator speed and load response.
